### DISCUSSION TITLE
Adapt it for charcoal smoker: Probes > PID > Fan

### DISCUSSION THREAD

**Discussion by waczny** [nebhead/PiFire#63] [Ideas]
I'm thinking to try to adapt this project, so you can use it for a regular charcoal smoker too. 
90% should be already in place with this awesome project here.
Technically you would have to change from the auger to a fan to controll the airflow, maybe a adjustable vent too.
I also thought about a lid control, once you open the lid and the temp drops too hard/ or a sensor reads lid open, the controller wont adjust for some minutes. This would be to prevent overshooting of the temp. 
I will start looking into it. 
What do you think of that? I hope I'll find some followers 
Best regards 
waczny

**Comment by nebhead** [comment]
Hey @waczny!  Just replied to your comment on YouTube.  I think this community can help where needed.  Just let us know where you need a little guidance.  I have seen others in the past who wanted to something similar (even adapting for a pizza oven!) so you are probably in good company.  In one of our other discussions, we were considering more granular fan control utilizing a DC PWM fan for future board designs.  This could be what you are looking for, with regards to controlling airflow in your charcoal grill.  

Also, lid control has been a topic of discussion for a while here too.  Thus far, it's not clear how we would implement this in hardware, given the heat of the lid etc.  But it's definitely on our minds.

**Reply by waczny** [reply]
Nice to receive an answer that fast.
I'm more in the industry and usually work with PLCs from Siemens. With all the Arduinos/PIs/etc. I have basic knowledge. I'm capable to modify/alter programs or write very simple ones from scratch, but something like this project would be too much for me as of right know. I always have big and great ideas but then I fail because I don't know how to start.

Yes I completly agree, a PWM controlled fan would be the best solution.
To controll the lid you could use a mechanically acutated switch or inductive sensor. The only problem is to resist the heat.
Another idea would be to run a glas fiber into the grill and monitor an open lid with the change of the ambient brightness, problem here would be dirt on the "lens".
I have to make some tests to see how hot a grill can get on the outer surface. 
I will also start to try using your current setup and just replace the auger with a fan to see what it does. 

Thank you so far I hope we (the community) can figure something out. 
BR waczny

**Reply by faisalhusain007** [reply]
Hey there @waczny, I think it might be easier for you to go the [heatermeater](https://github.com/CapnBry/HeaterMeter/wiki), it is a mature project and  works really well for fan pwm control. Uses a hybrid combo of a pi (can use the zero) and a MCU (Atmel can't remember but really cheap) The pi only gives you wireless access to control and plot temps etc. Would love for you to stay and contribute, but its worth a look at. I have a Masterbuilt Gravity 560, it uses a pwm fan control in a very similar way to the heater meter. The Masterbuilt has switches that turn the fan down when the lid is lifted , and a further set connected in  series where the charcoal  / fuel chamber  is. These switch the fan off completely  for safety when opened. The switches are available [here](https://prosmokebbq.co.uk/products/masterbuilt-replacement-door-switch?variant=39374468284510) probably way cheaper where you guys are. The heater meter doesn't use a  switch it detects a sudden change in temperature (user defined) and goes in to lid opened mode for a given time (User defined but default is  2 or 3 minutes can't remember.) 

Hope that helps.

I forgot to mention the MB560 working temp range is 150-700f so I guess the switches are good for that range, personally I have never gone over 650f yet and so far not had any issues with my unit.

**Reply by waczny** [reply]
Hi,
I already looked into the heater meater but I have to say I like the UI of this project way better and I started to put the components in the basket and ended up with over 70$ only for electric components without the PCBs and I woould have order stuff from 3 different stores (Mauser/Newark/DigiKey) because none of them have all at once and the HeaterMeter store is sold out. Also if I look at both projects this one seems to be still active and the other doesn’t have that much activity anymore.

**Reply by nebhead** [reply]
> I have a Masterbuilt Gravity 560, it uses a pwm fan control in a very similar way to the heater meter. The Masterbuilt has switches that turn the fan down when the lid is lifted , and a further set connected in series where the charcoal / fuel chamber is. These switch the fan off completely for safety when opened. The switches are available [here](https://prosmokebbq.co.uk/products/masterbuilt-replacement-door-switch?variant=39374468284510) probably way cheaper where you guys are. The heater meter doesn't use a switch it detects a sudden change in temperature (user defined) and goes in to lid opened mode for a given time (User defined but default is 2 or 3 minutes can't remember.)

I'm intrigued by this switch that you have linked, but also I like the simplicity of the lid-open detection based on temperature swings.  That might be harder to detect, but also worth a look. 

And for what it's worth, I would agree with @faisalhusain007 that Heatermeter is a great project to look at.  In fact, it's part of the inspiration for this project, and where I got the probe temperature algorithms from.  But still happy to have you here, and am curious to see what you find with applying this PWM to the fan.  I recently stumbled accross the PID tuning portion of the Heatermeter project and found it pretty interesting.  [(link here)](https://github.com/CapnBry/HeaterMeter/wiki/PID-Tuning)  Might be worth a read through to see if any of the PID values need to be tweaked.

**Reply by faisalhusain007** [reply]
Since we are designing the next pcb and phase of this project, I think we factor in both and give the choice to the user, the temperature swing works great when it's tuned to the grill, the switch route works well but need to mount and install on the grill.  Personally I prefer the temp swing as it's fairly predictable on my grill the ambient temperature is the same all year round over here.

**Reply by waczny** [reply]
Yeah i was finally able to order all parts for the Heatermeter, but I still keep my eye on here and try to support.
